recipes/cassandra-cpp-driver/all/patches/2.15.3/fix-cmake.patch

```diff
@@ -0,0 +1,466 @@
+-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}  # for shared library
+-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}) # for static library
+-  if(CASS_INSTALL_PKG_CONFIG)
+-    if(NOT WIN32)
+-      if(PKG_CONFIG_FOUND)
+-        configure_file("${PROJECT_SOURCE_DIR}/packaging/cassandra.pc.in" "cassandra.pc" @ONLY)
+-        install(FILES "${CMAKE_CURRENT_BINARY_DIR}/cassandra.pc"
+-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+-      endif()
+-    endif()
+-  endif()
+-  if(WIN32)
+-    install(FILES $<TARGET_PDB_FILE:cassandra>
+-      DESTINATION "${INSTALL_DLL_EXE_DIR}"
+-      OPTIONAL)
+-  endif()
+-endif()
+-
+-if(CASS_BUILD_STATIC)
+-  install(TARGETS cassandra_static
+-    RUNTIME DESTINATION ${INSTALL_DLL_EXE_DIR}  # for dll/executable/pdb files
+-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}  # for shared library
+-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}) # for static library
+-  if(CASS_INSTALL_PKG_CONFIG)
+-    if(NOT WIN32)
+-      if(PKG_CONFIG_FOUND)
+-        configure_file("${PROJECT_SOURCE_DIR}/packaging/cassandra_static.pc.in" "cassandra_static.pc" @ONLY)
+-        install(FILES "${CMAKE_CURRENT_BINARY_DIR}/cassandra_static.pc"
+-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+-      endif()
+-    endif()
+-  endif()
+-endif()
++# Install the library
++install(TARGETS ${PROJECT_LIB_NAME_TARGET}
++  RUNTIME DESTINATION ${INSTALL_DLL_EXE_DIR}  # for dll/executable/pdb files
++  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}  # for shared library
++  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}) # for static library
+ 
+ #-------------------
+ # Uninstall target
+diff --git a/src/third_party/curl/CMakeLists.txt b/src/third_party/curl/CMakeLists.txt
+index 63f391752..8c00dd203 100644
+--- a/src/third_party/curl/CMakeLists.txt
++++ b/src/third_party/curl/CMakeLists.txt
+@@ -3,16 +3,8 @@ file(GLOB SOURCES *.cpp *.hpp)
+ source_group("Source Files\\" REGULAR_EXPRESSION ".*\\.c(pp)?")
+ source_group("Header Files\\" REGULAR_EXPRESSION ".*\\.h(pp)?")
+ 
+-if(CASS_BUILD_SHARED)
+-  add_library(curl_hostcheck OBJECT ${SOURCES})
+-  set_target_properties(curl_hostcheck PROPERTIES 
+-    POSITION_INDEPENDENT_CODE ON 
+-    FOLDER "Driver/Dependencies")
+-endif()
+-
+-if(CASS_BUILD_STATIC)
+-  add_library(curl_hostcheck_static OBJECT ${SOURCES})
+-  set_target_properties(curl_hostcheck_static PROPERTIES 
+-    FOLDER "Driver/Dependencies")
+-endif()
++add_library(curl_hostcheck OBJECT ${SOURCES})
++set_target_properties(curl_hostcheck PROPERTIES 
++  POSITION_INDEPENDENT_CODE ON 
++  FOLDER "Driver/Dependencies")
+ 
+diff --git a/src/third_party/hdr_histogram/CMakeLists.txt b/src/third_party/hdr_histogram/CMakeLists.txt
+index 41e4bab5b..890df4e4b 100644
+--- a/src/third_party/hdr_histogram/CMakeLists.txt
++++ b/src/third_party/hdr_histogram/CMakeLists.txt
+@@ -3,15 +3,8 @@ file(GLOB SOURCES *.cpp *.hpp)
+ source_group("Source Files\\" REGULAR_EXPRESSION ".*\\.c(pp)?")
```

Comment:
Can we use a conan `hdr_histogram`?

Reply:
No we can not. Cassandra uses patched hdr_histogram version.